pkg/provider/consulcatalog/config.go

```diff
@@ -131,6 +131,21 @@ func (c *connectCert) isReady() bool {
 	return len(c.root) > 0 && c.leaf.cert != "" && c.leaf.key != ""
 }
 
+func (c *connectCert) Equal(other *connectCert) bool {
+	if other == nil {
+		return false
+	}
+	if len(c.root) != len(other.root) {
+		return false
+	}
+	for i, v := range c.root {
+		if v != other.root[i] {
+			return false
+		}
+	}
+	return c.leaf == other.leaf
```

Comment:
Shouldn't we also check c.err VS other.err ?

Reply:
Good question, the way it is used it shouldn't matter. Because errors are handled differently; I'll see what I can do.

Reply:
Mhm, according to the test failures my error comparison is not good; any hints on how to do that properly in Go?

Reply:
Is this still an open question? If yes, which test are you talking about? (the tests from the CI look all green at the time I'm writing this).
In any case, my initial suggestion was more theoretical. I think since the err is not a real property of the connectCert (we're sneaking it in as a convenience), it would be fine not to take it into account for the equality. So feel free to ignore it out if it's causing you any problem.

Reply:
ah I see now, the linter is choking on it, bleh.

```
---> Making bundle: validate-lint (in /go/src/github.com/traefik/traefik/script)
pkg/provider/consulcatalog/config.go:149:33: comparing with == will fail on wrapped errors. Use errors.Is to check for a specific error (errorlint)
	return c.leaf == other.leaf && c.err == other.err
	                               ^
Makefile:106: recipe for target 'validate' failed
```

Oh well, you could either apply the suggestion (use errors.Is instead of direct comparison), or just drop the error comparison altogether.